### DISCUSSION TITLE
How to permit only authorized users to do one of the CRUD operations?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#634] [Q&A]
I want to allow only authorized users to perform create, update or delete a record. How do I?

**Comment by ganigeorgiev** [comment]
You have to define the following filter in your collection create, update and delete API rules:

```
@request.user.id != ""
```

You could find other examples here - https://pocketbase.io/docs/manage-collections/#rules-filters-examples

**Reply by Mr-M1M3** [reply]
@ganigeorgiev wouldn't it allow all the registered users?
Let's say, I have a collection called post. And I want to update a record of post collection. But, the user who created that post can update this record. How do I do that?

**Reply by ganigeorgiev** [reply]
You have 2 options:

1. Add `user` type field, eg. `userId`, to your `posts` collection and for the API rules you could use something like:
    ```
    userId = @request.user.id
    ```

2. You can also create a `relation` type field pointing to the `profiles` collection (eg. author) and use its `userId` reference in the access checks:
    ```
    author.userId = @request.user.id

    # or direct compare with the profile
    author = @request.user.profile.id
    ```
   The `relation` type field allow also expansion when calling for example: `client.records.getList("posts", 1, 50, { expand: "author" })`.


**_Please note that this 2 type of fields will be merged soon into one to avoid the ambiguities and simplify the users management per #376._**

**Reply by Mr-M1M3** [reply]
@ganigeorgiev Hello again! I tried the first one. It seemed like it worked. However, I cannot create a record as I need an `userId` to create a record. Any solution please?

**Reply by ganigeorgiev** [reply]
@Mr-M1M3 The `userId` is not assigned by default. You need to pass it as body param alongside your other fields. For example:

```
{
    "userId": client.authStore.model?.id,
    "name": "test",
    ...
}
```

_footnote: Make sure that the `userId` field is marked "Required" or change the filter to `@request.user.id != "" && userId = @request.user.id`._
